_posts/2020-07-06-concurrency-costs.md

```diff
@@ -557,15 +557,15 @@ The _~Cost_ column is a _very_ approximate estimate of the cost of each "occurre
 
 The _Perf Event_ column lists a Linux `perf` event that you can use to count the number of times the operation associated with this level occurs, i.e., the thing that is slow. For example, in level 1, you count atomic operations using the `mem_inst_retired.lock_loads` counter, and if you get three counts per high level operation, you can expect roughly 3 x 10 ns = 30 ns cost. Of course, you don't necessarily need perf in this case: you can inspect the assembly too.
 
-The _Local_ column records whether the behavior of this level is _core local_. If yes, it means that operations on different cores complete independently and don't compete and so the performance scales with the number of cores. If not, there is contention or serialization, so the throughput of the entire system is often limited, regardless of how many cores are involved. For example, only one core at time performs an atomic operation on a cache line, so the throughput of the whole system is fixed and the throughput per core decreases as more cores become involved.
+The _Local_ column records whether the behavior of this level is _core local_. If yes, it means that operations on different cores complete independently and don't compete and so the performance scales with the number of cores. If not, there is contention or serialization, so the throughput of the entire system is often limited, regardless of how many cores are involved. For example, only one core at a time performs an atomic operation on a cache line, so the throughput of the whole system is fixed and the throughput per core decreases as more cores become involved.
 
 The _Key Characteristic_ tries to get across the idea of the level in one bit-sized chunk.
 
 | Level |  Name | ~Cost (ns) | Perf Event | Local | Key Characteristic |
 | ---   | ------| -------------:|:----------------:|:----:| -- |
 | 0 | Vanilla | low | depends | **Yes**{:.yes.yesno} | No atomic instructions or contended accesses at all |
 | 1 | Uncontended Atomic | 10 | `mem_inst_retired.` `lock_loads` | **Yes**{:.yes.yesno} | Atomic instructions without contention |
-| 2 | True Sharing | 40 - 400 | `mem_load_l3_hit_retired.` `xsnp_hitm` | **No**{:.no.yesno} | Contented atomics or locks |
+| 2 | True Sharing | 40 - 400 | `mem_load_l3_hit_retired.` `xsnp_hitm` | **No**{:.no.yesno} | Contended atomics or locks |
```

Comment:
lol

Reply:
Yeah that one made me smile. We should all strive to write contented software!